### ISSUE TITLE
[solc] Throws ICE "Invalid conversion to calldata type"

### ISSUE THREAD

**Issue by bshastry** [ethereum/solidity#8501]
## Description

```
contract C {
  uint[] m_data_dyn;
  function s(uint[6][] calldata b) external {
    m_data_dyn = b[1];
  }
}
```

throws

https://github.com/ethereum/solidity/blob/435c9dae718eeef645edb715293c4c687240ee5e/libsolidity/codegen/CompilerUtils.cpp#L975-L980

## Environment

- Compiler version: latest develop

## Steps to Reproduce

```
$ solc <input.sol>
Internal compiler error during compilation:
/home/bhargava/work/github/solidity/libsolidity/codegen/CompilerUtils.cpp(980): Throw in function void solidity::frontend::CompilerUtils::convertType(const solidity::frontend::Type &, const solidity::frontend::Type &, bool, bool, bool)
Dynamic exception type: boost::exception_detail::clone_impl<solidity::langutil::InternalCompilerError>
std::exception::what: Invalid conversion to calldata type.
[solidity::util::tag_comment*] = Invalid conversion to calldata type.
```

**Comment by axic** [comment]
Is not reproducible anymore. Not sure which PR fixed it.

**Comment by chriseth** [comment]
Yep, first fixed version is 0.6.5.
